test/ibmq/test_experiment.py

```diff
@@ -753,3 +773,30 @@ def _create_analysis_result(self):
         self.provider.experiment.upload_analysis_result(new_result)
         self.results_to_delete.append(new_result.uuid)
         return new_result
+
+    @classmethod
+    def _create_initial_test_data(cls):
+        """Creates an initial push of test data."""
+        # Generate 3 pieces of data.
+        for _ in range(3):
+            # Create an experiment
+            exp = Experiment(  # pylint: disable=unexpected-keyword-arg
+                provider=cls.provider,
+                backend_name=cls.BACKEND_NAME,
+                experiment_type='test',
+                tags=['qiskit-test', 'qiskit-test-1'],  # 2 tags for multi-tag checks
+                start_datetime=datetime.now(),
+                share_level=ExperimentShareLevel.PUBLIC,
+                notes='test notes'
+            )
+            cls.provider.experiment.upload_experiment(exp)
```

Comment:
Should be creating the experiments per the helper function (`_create_experiment`) that is tracking the experiments and deleting them on tearDown otherwise these CI-generated experiments are hanging out in the database between runs:

![image](https://user-images.githubusercontent.com/3237936/125961626-306cd3f4-c772-4c56-aeb5-5ef2dc052c10.png)


That's not really ideal for a CI/test system.

Reply:
The target of this function was to create persistent test records, in the event the user didn't have any to start with. This would make it so the initial calls to get `qiskit-test` records in `setupClass()` would be non-empty.

Reply:
> 
> 
> The target of this function was to create persistent test records, in the event the user didn't have any to start with. This would make it so the initial calls to get `qiskit-test` records in `setupClass()` would be non-empty.

We don't really _need_ persistent test records though right? Otherwise on the service side I'm going to have to implement a cron job that goes through and deletes old `qiskit-test` experiments. It shouldn't be that big a deal as part of the test class setup to assume there aren't any records and just create some test data like this is already doing.

Reply:
I agree that we don't need persistent test records. From what I understand, this test suite [was already dependent on them](https://github.com/Qiskit/qiskit-ibmq-provider/blob/master/test/ibmq/test_experiment.py#L46). It depends on which you prioritize:
Pros: Save creating jobs each time the test suite runs
Cons: Jobs remain in the user's job list

I personally don't think this is a harm to the user experience (i.e wouldn't require a cron job deleting):
1. the only users to see these experiments are those running these tests (developers)
2. these experiments are run only once and should fall down a user's experiments list very quickly

@mriedem If you still are thinking persistent test records should be removed, I'll remove them.

Reply:
> 
> 
> I agree that we don't need persistent test records. From what I understand, this test suite [was already dependent on them](https://github.com/Qiskit/qiskit-ibmq-provider/blob/master/test/ibmq/test_experiment.py#L46). It depends on which you prioritize:
> Pros: Save creating jobs each time the test suite runs
> Cons: Jobs remain in the user's job list
> 
> I personally don't think this is a harm to the user experience (i.e wouldn't require a cron job deleting):
> 
>     1. the only users to see these experiments are those running these tests (developers)
> 
>     2. these experiments are run only once and should fall down a user's experiments list very quickly
> 
> 
> @mriedem If you still are thinking persistent test records should be removed, I'll remove them.

Yes the tests were originally written assuming there were experiments in the database it had access to. I wouldn't say that's the best way to write tests though. CI should assume a clean environment to start and try to clean up after itself.

Also, these aren't IQX jobs, they are resultsdb experiments - different resources. The experiments resources created in these tests aren't actually running any jobs on the backend systems so they are quick to create and delete.

>the only users to see these experiments are those running these tests (developers)

This isn't true, the experiments created here are public which means anyone in the staging environment in IQX has access to them:

https://www-dev.quantum-computing.ibm.com/experiments?date_interval=last-90-days

![image](https://user-images.githubusercontent.com/3237936/126334761-c531a7c8-422e-486f-b014-8a350fe3dd02.png)

That's not a huge deal but it could make someone wonder why there are these CI-generated public experiments hanging out.

So yeah my preference is that the tests should create what they need and clean up after themselves.

Reply:
Updated! Appreciate the reply and the patience.
`cls.analysis_results` and `cls.experiments` both pull from the test objects and `tearDownClass` will remove them!